pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -306,10 +306,6 @@ func validateParametersInTaskMatrix(matrix []Param) (errs *apis.FieldError) {
 		if param.Value.Type != ParamTypeArray {
 			errs = errs.Also(apis.ErrInvalidValue("parameters of type array only are allowed in matrix", "").ViaFieldKey("matrix", param.Name))
 		}
-		// results are not yet allowed in parameters in a matrix - dynamic fanning out will be supported in future milestone
```

Comment:
would it make sense to add a test case for a valid param in a matrix referencing a result?

Reply:
`validateParametersInTaskMatrix` is an unexported and untested function - it is used in `validateMatrix` which is tested - see `Test_validateMatrix` in `pipeline_types_test.go`. those tests used to fail because consuming results was invalid - and now we've removed the errors because the cases are valid.

Reply:
got it!